sympy/sets/fancysets.py

```diff
@@ -766,6 +766,15 @@ def _sup(self):
     def _boundary(self):
         return self
 
+    def as_relational(self, symbol):
+        """Rewrite a FiniteSet in terms of equalities and logic operators. """
+        inf_end_points = [S.NegativeInfinity, S.Infinity]
+        if self.start in inf_end_points or self.stop in inf_end_points:
+            from sympy.functions.elementary.integers import floor
+            return And(symbol >= self.start, symbol < self.stop, Eq(floor(symbol), symbol))
+        else:
+            return Or(*[Eq(symbol, elem) for elem in self])
```

Comment:
Do you really want to return a 1000-arg Or for Range(1000)?

How about the following:
```
def as_relational(self, x):
    i = (x - (self.inf if self.inf.is_finite else self.sup))/self.step
    return And(
        Eq(i, floor(i)),
        x >= self.inf if self.inf in self else x > self.inf,
        x <= self.sup if self.sup in self else x < self.sup)
```

Reply:
> Do you really want to return a 1000-arg Or for Range(1000)?

Ah! point, well then this modification should be done for `FiniteSet` too, as it is based on the same approach.

Reply:
A finite set has no guarantee of disrete step size or uniform type containment. The only sets that could benefit from the same logic would have to have the same internal structure as a Range.